src/pages/signin/SignInPage.js

```diff
@@ -186,6 +192,9 @@ function SignInPage({credentials, account, isInModal, activeClients, preferredLo
     if (shouldInitiateSAMLLogin) {
         setHasInitiatedSAMLLogin(true);
         Navigation.isNavigationReady().then(() => Navigation.navigate(ROUTES.SAML_SIGN_IN));
+    } else if (hasInitiatedSAMLLogin) {
+        // Return early because we're already navigating to a different page
+        return;
```

Comment:
Hmm, I'm not getting that blank page while testing @fedirjh - are you experiencing that only on mobile or also on web? Also if we remove the `hasInitiatedSAMLLogin` logic we'll get the `[warn] SignInPage in unexpected state! - ""` error when the user signs in using the SAML required flow. 

What about something like: 

```diff
-function getRenderOptions({hasLogin, hasValidateCode, account, isPrimaryLogin, isUsingMagicCode, hasInitiatedSAMLLogin, isClientTheLeader}) {
+function getRenderOptions({hasLogin, hasValidateCode, account, isPrimaryLogin, isUsingMagicCode, hasInitiatedSAMLLogin, isClientTheLeader, hasAccountError}) {
     const hasAccount = !_.isEmpty(account);
     const isSAMLEnabled = Boolean(account.isSAMLEnabled);
     const isSAMLRequired = Boolean(account.isSAMLRequired);
@@ -109,7 +109,7 @@ function getRenderOptions({hasLogin, hasValidateCode, account, isPrimaryLogin, i
         Session.clearSignInData();
     }
 
-    const shouldShowLoginForm = isClientTheLeader && !hasLogin && !hasValidateCode;
+    const shouldShowLoginForm = hasAccountError || isClientTheLeader && !hasLogin && !hasValidateCode;
     const shouldShowEmailDeliveryFailurePage = hasLogin && hasEmailDeliveryFailure && !shouldShowChooseSSOOrMagicCode && !shouldInitiateSAMLLogin;
     const isUnvalidatedSecondaryLogin = hasLogin && !isPrimaryLogin && !account.validated && !hasEmailDeliveryFailure;
     const shouldShowValidateCodeForm =
@@ -148,6 +148,7 @@ function SignInPage({credentials, account, isInModal, activeClients, preferredLo
     const [hasInitiatedSAMLLogin, setHasInitiatedSAMLLogin] = useState(false);
 
     const isClientTheLeader = activeClients && ActiveClientManager.isClientTheLeader();
+    const hasAccountError = !_.isEmpty(account.errors);
 
     useEffect(() => Performance.measureTTI(), []);
     useEffect(() => {
@@ -187,12 +188,13 @@ function SignInPage({credentials, account, isInModal, activeClients, preferredLo
         isUsingMagicCode,
         hasInitiatedSAMLLogin,
         isClientTheLeader,
+        hasAccountError,
     });
 
     if (shouldInitiateSAMLLogin) {
         setHasInitiatedSAMLLogin(true);
         Navigation.isNavigationReady().then(() => Navigation.navigate(ROUTES.SAML_SIGN_IN));
-    } else if (hasInitiatedSAMLLogin) {
+    } else if (hasInitiatedSAMLLogin && !hasAccountError) {
         // Return early because we're already navigating to a different page
         return;
     }
```

Reply:
Sorry for the delay here - had some simulator issues and then was out of office for the US holiday weekend. I'm unable to reproduce the behavior you're seeing with the blank screen, but I do see an issue where the user gets redirected to the oldDot error page instead. That's because of a re-routing issue on the backend, which I'll push a PR up shortly to fix. 

https://github.com/Expensify/App/assets/16747822/b99ff77f-be32-4d7d-b3fb-fb47bdeb234b

I'd rather get this PR out now so that the success flow works on production, and then do a follow up PR to more accurately handle failed sign-in cases. Given that, I can revert my commit that added the following logic: 

```
    } else if (hasInitiatedSAMLLogin) {
        // Return early because we're already navigating to a different page
        return;
```

How does that sound?

Reply:
I lied, I was able to reproduce it. Updated some logic and it should be working now 🤞 🍀

Reply:
Found one small bug :bug: adressing it now